### PR TITLE
Pass keyword arguments to `AutoModelForCausalLM`

### PR DESCRIPTION
**Pull request by arunpatro**
Without `trust_remote_code` set to `true`, we cannot run models like https://huggingface.co/mosaicml/mpt-7b and falcon. HF mentions to add this flag. See here -> https://huggingface.co/docs/transformers/custom_models#using-a-model-with-custom-code

### CODE REVIEW

**Comment by rlouf** [comment]
Should we allow users to pass this keyword argument when initialising the model and use whatever default the `transformers` library uses? This looks like a legit security concern.

**Comment by arunpatro** [comment]
> Should we allow users to pass this keyword argument when initialising the model and use whatever default the `transformers` library uses? This looks like a legit security concern.

I agree with the security concern, but users would definitely want to run latest models like falcon, etc. We can raise a meaningful error too. 

There are other keywords that are relevant which can be passed to the hf model like quantization flags. We should support these features and more `kwargs` because local inference is preffered by community and also because apple silicon is a beast.

**Comment by rlouf** [comment]
> I agree with the security concern, but users would definitely want to run latest models like falcon, etc. We can raise a meaningful error too. 

Unless we have a good reason to not do so we should use the same defaults as the `transformers` library. For [this blog post](https://huggingface.co/blog/falcon):

```python
from transformers import AutoTokenizer
import transformers
import torch

model = "tiiuae/falcon-7b-instruct"

tokenizer = AutoTokenizer.from_pretrained(model)
pipeline = transformers.pipeline(
    "text-generation",
    model=model,
    tokenizer=tokenizer,
    torch_dtype=torch.bfloat16,
    trust_remote_code=True,
    device_map="auto",
)
```

Beyond the security concerns (which should be taken seriously) the interface should aim to be non-surprising for users of the `transformers` library. 

> There are other keywords that are relevant which can be passed to the hf model like quantization flags. We should support these features and more `kwargs` because local inference is preffered by community and also because apple silicon is a beast. 

We definitely should. I think it shouldn't take a lot of extra work to do it in this PR, at least for the most used options.

File: outlines/models/hf_transformers.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def call_model_generate_method(
     prompt = list(prompt)
 
     tokenizer = AutoTokenizer.from_pretrained(model_name, padding_size="left")
-    model = AutoModelForCausalLM.from_pretrained(model_name)
+    model = AutoModelForCausalLM.from_pretrained(model_name, trust_remote_code=True)
```
Comment (rlouf) [inline outlines/models/hf_transformers.py:96]: We probably shouldn't set this to `True` by default

**Review by rlouf — CHANGES_REQUESTED** [review]

File: outlines/models/hf_transformers.py (hunk 1 above)
Comment (dgerlanc) [reply outlines/models/hf_transformers.py:96]: This 100% shouldn't be set to `True` by default. As Rémi said, we want to respect the library defaults here.

**Comment by arunpatro** [comment]
What about allowing users to do something like `outlines.trust_remote_code = True` at the top like we use for disabling cache?

**Comment by rlouf** [comment]
The best option imo is to use`**kwargs` for passthrough arguments.

**Comment by rlouf** [comment]
Being implemented in #139, closing.

**Pull request closed without merging**